rosbag2_test_common/include/rosbag2_test_common/subscription_manager.hpp

```diff
@@ -127,13 +127,56 @@ class SubscriptionManager
     return matched;
   }
 
-  std::future<void> spin_subscriptions()
+  /// @brief Wait until publishers will be connected to the subscriptions or timeout occur.
+  /// @param topic_names List of topic names
+  /// @param timeout Maximum time duration during which discovery should happen.
+  /// @param n_publishers_to_match Number of publishers each subscription should have for match.
+  /// @return true if subscriptions have specified number of publishers, otherwise false.
+  bool spin_and_wait_for_matched(
+    const std::vector<std::string> & topic_names,
+    std::chrono::duration<double> timeout = std::chrono::seconds(10),
+    size_t n_publishers_to_match = 1)
+  {
+    // Sanity check that we have valid input
+    if (topic_names.empty()) {
+      throw std::invalid_argument("List of topic names is empty");
+    }
+    for (const auto & topic_name : topic_names) {
+      if (subscriptions_.find(topic_name) == subscriptions_.end()) {
```

Comment:
This is a nitpick, but I find using count to be more readable here.  But feel free to ignore this comment if you like `find` better:

```suggestion
      if (subscriptions_.count(topic_name) == 0) {
```